logstash-core/spec/logstash/pipeline_spec.rb

```diff
@@ -529,26 +529,45 @@ class TestPipeline < LogStash::Pipeline
     end
   end
 
-  context "when collecting metric in the pipeline" do
+  context "when collecting metrics in the pipeline" do
     subject { described_class.new(config, { :metric => metric, :pipeline_id => pipeline_id }) }
     let(:pipeline_id) { :main }
     let(:metric) { LogStash::Instrument::Metric.new }
     let(:number_of_events) { 1000 }
+    let(:multiline_id) { "my-multiline" }
+    let(:multiline_id_2) { "my-multiline_2" }
+    let(:dummy_output_id) { "my-dummyoutput" }
+    let(:generator_id) { "my-generator" }
     let(:config) do
       <<-EOS
-      input { generator { count => #{number_of_events}} }
+      input {
+        generator {
+           count => #{number_of_events}
+           id => "#{generator_id}"
+        }
+      }
       filter {
          multiline {
+              id => "#{multiline_id}"
               pattern => "hello"
               what => next
           }
+          multiline {
+               id => "#{multiline_id_2}"
+               pattern => "hello"
+               what => next
+           }
+      }
+      output {
+        dummyoutput {
+          id => "#{dummy_output_id}"
+        }
       }
-      output { dummyoutput {} }
       EOS
     end
-    let(:dummyoutput) { DummyOutput.new }
+    let(:dummyoutput) { DummyOutput.new({ "id" => dummy_output_id }) }
 
-    before do
+    before :each do
       allow(DummyOutput).to receive(:new).with(any_args).and_return(dummyoutput)
       allow(LogStash::Plugin).to receive(:lookup).with("input", "generator").and_return(LogStash::Inputs::Generator)
```

Comment:
are this necessary? I mean if DummyOutput inherits from plugin should be able to get real plugins, isn't?